os/mm/mm_heap/mm_malloc.c

```diff
@@ -108,6 +112,18 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size, mmaddress_t caller_
 FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
 #endif
 {
+#ifdef CONFIG_BOOTMEM
+	if (g_bootmem) {
+		void *ret = g_bootmem;
+		size = MM_ALIGN_UP(size);
```

Comment:
Is there any specific reason for providing 4 byte alignment for every allocated pointer. I don't see any need of it.